### ISSUE TITLE
I can't configure a mod item in Auto Vanilla Hands

### ISSUE THREAD

**Issue by peow** [tr7zw/FirstPersonModel#424]
i'm using a mod that adds a cave map, called Cave Survey (https://modrinth.com/mod/cave-survey), but it doesn't work with FirstPersonModel + NotEnoughAnimations, so I disabled it in the settings, but it seems that the settings have no effect, I put both the empty map and full map but Auto Vanilla Hands still does not work

Cave Map without FirstPersonModel:
![2024-01-29_12 05 15](https://github.com/tr7zw/FirstPersonModel/assets/105888751/5e9888ed-a273-427f-bf8f-604ee70e94bf)

Cave Map with FirstPersonModel configured to use AutoVanillaHands:
![2024-01-29_12 05 27](https://github.com/tr7zw/FirstPersonModel/assets/105888751/7b1e97f9-e998-42d2-9396-c233ebecf6d1)

---------------

my config file:

{
  "configVersion": 2,
  "enabledByDefault": true,
  "xOffset": 0,
  "sneakXOffset": 0,
  "sitXOffset": 0,
  "renderStuckFeatures": false,
  "vanillaHands": false,
  "doubleHands": false,
  "autoVanillaHands": [
    "twilightforest:filled_ore_map",
    "twilightforest:filled_magic_map",
    "twilightforest:filled_maze_map",
    "antiqueatlas:antique_atlas",
    "cave-survey:filled_cave_map",
    "cave-survey:cave_map",
    "polaroidcamera:camera"
  ],
  "autoToggleModItems": [
    "exposure:camera",
    "cave-survey:filled_cave_map",
    "cave-survey:cave_map",
    "polaroidcamera:camera"
  ]
}

* "polaroidcamera:camera" (other mod item) works fine, just cave maps don't

* im using Fabric 1.20.1, FirstPersonModel v2.3.2

**Comment by tr7zw** [comment]
Did you restart the game after changing the config? It's not live reloading.

**Comment by Barerock** [comment]
My config file is resetting itself whenever I add anything to it, I will have to set it to read only.

**Comment by tr7zw** [comment]
> My config file is resetting itself whenever I add anything to it, I will have to set it to read only.

No, that's means the json is invalid or you had the game running and then tried to modify the config in there.

**Comment by Barerock** [comment]
> No, that's means the json is invalid or you had the game running and then tried to modify the config in there.

Why would the json be invalid? It's definitely reading the file and modifying it when not in read only mode. I also close the game completely to edit this mod's config, so that's not the issue.

**Comment by tr7zw** [comment]
If it rests when the game starts = the file is invalid. Not the values in there, the formatting.

**Comment by Barerock** [comment]
I know exactly what I did wrong, disregard my input to this issue. I will say that the CGM autovanillahands is janky af even after your funny fix. Definitely turning that off.

**Comment by tr7zw** [comment]
I mean, it works as I described it #357.

**Comment by Barerock** [comment]
Oh it very much works, it's just not at all natural, which is extremely unfortunate. But who needs iron sights anyway?

**Comment by tr7zw** [comment]
but the sights work?

**Comment by Barerock** [comment]
Before I leave this issue to the main topic, would it be possible to implement a whitelisted function for the X-Y-Z visual offset? Like the autovanillahands list, a list of things with custom XYZ views to them? Currently the sights are functional with severe sight adjustment, but those edits make the game near unplayable for every other instance. It'd be neat to be able to individually edit an item's sight line.

Yes, it correctly applies autovanillahands, however the first person model retains its arms like normal and the rotation of CGM guns makes it so that the left arm is always visible, creating an uncanny feeling. So we've subtracted from the 4 arms to 3 arms lol.

**Comment by tr7zw** [comment]
You added the weapons to the wrong part of the config. Add them to autoToggleModItems instead of vanilla hands. Then you don't have this problem.

**Comment by Barerock** [comment]
oooh, I see.

**Comment by tr7zw** [comment]
Or wait huh. It's clearly too late for my mind. Also no clue what you mean with XYZ offsets. Nothing anywhere has any special offsets.

**Comment by Barerock** [comment]
The visual offset you can create in the config menu ingame. Screenshot incoming once I start it up.

**Comment by Barerock** [comment]
![image](https://github.com/tr7zw/FirstPersonModel/assets/38869993/806c550f-1031-4b62-95a0-627eb3a88ede)

**Comment by tr7zw** [comment]
> The visual offset you can create in the config menu ingame. Screenshot incoming once I start it up.

huh? Thats the model offset and only the X position in case a custom model or armor is in the way. What does that have to do with items?

**Comment by Barerock** [comment]
It fixes the sightline on the ironsights of CGM guns when aiming lol

**Comment by tr7zw** [comment]
Just no. CGM is NOT a 3rd person compatible mod. Thats it. It doesn't work. Wasn't intended to work by either Mr Crayfish nor me.

**Comment by Barerock** [comment]
To resolve this to a more functional system, a whitelist for specific items that need the entire arm and item model shifted whichever way to see them properly, rather than the actual sightline (I believe that'd just make a gun inaccurate), configurable by the player to avoid unnecessary work on your part if you were to implement such a system.

**Comment by Barerock** [comment]
Simply put, being able to mark an item to be shifted a set of directions when held.

**Comment by Barerock** [comment]
I don't know java, I'm just an idea guy, but I do know says this _should_ be possible. No idea how hard it would be though.

**Comment by tr7zw** [comment]
I'll be bland. I dont get why people install gun mods into mc. I don't see the appeal.
It's great that Mr.Crayfish added a 3rd person animation(better than most mods these days), but it wasn't intended to be "right" in relation to the camera/head (same with any mojang animation). 
> function for the X-Y-Z visual offset?

Now think about that. How should this look like? Should the entire player float and your eyes come out of the shoulder because you had to move up the model? Should the item be floating outside of the arms to line up(which wouldn't even work without hooking CGM)? Should the camera be moved, causing anticheat issues and also move the camera into the torso? Also when rotating the camera a simple XYZ offset wouldn't work for both cases since the camera pos is fixed in place when rotating, while the player model/player head is rotating on a circle.
OR just use the firstperson model by Crayfish that is intended to work correctly and does work perfectly fine(via vanilla hands/autoToggleModItems)(Or just don't use CGM and FPM at the same time)?
It just makes no sense to even try to sink time into this as it wont work, wont look great and be used by at best a few hundred people.

**Comment by Barerock** [comment]
Ngl, hating on people's preferences for things is silly in general. You're a mod author, not a mod curator, regardless of your status. I use both because A. they're cool B. I play much more difficult ways than most people C. I can - much like anyone else.

If you read what I wrote, moving the arms and the item would allow for compatibility between many items that simply don't sit right. The body isn't really visible during most playtime, and you're not gonna be holding most items more than a few seconds, meaning the awkwardness of any parts is outweighed by the functionality added. Configurability is part of why people use mods, and only applying what I said to one mod denigrates both of our intellects. I already said I don't know Java very much and like to pander ideas because who else will do it?

I'm out; you've clearly reached a point of exasperation that was completely unbeknownst to me until you started being rude.

**Comment by tr7zw** [comment]
It's just that I had this talk a few times. Moving the arms+ items is not an easy thing. And I develop this mod as a hobby, its not my job. So it's not in my interest to spend a bunch of time to try to get some mod combination to work that I personally have no interest in and that has a decent solution(using the vanilla view). What you are asking for sounds more like adding arms to the firstperson animations like this 
![grafik](https://github.com/tr7zw/FirstPersonModel/assets/5133936/a7972dac-3448-4638-b148-e7c5e9892c23)
But that's just not what this mod is about, that's an entirely different thing.

**Comment by tr7zw** [comment]
It was https://github.com/tr7zw/AnimatedFirstPerson , but canceled developing that mod as I suck too much in making decent animations, and Trainguy is working on a mod that looks way better.
https://gyazo.com/099244ee30f9835fdfbfdf3ea89b202b.mp4

**Comment by tr7zw** [comment]
https://discord.gg/CVnVP8Ff but sadly progress is slow.

**Comment by tr7zw** [comment]
Oh yea, please test with the latest build, also verify that in the log it found the correct item(and that the item name is right, for example ``cave-survey`` as the namespace is not valid? verify with F3+H).

**Comment by tr7zw** [comment]
You can see on the screenshot that its a ``cave-survey:filled_cave_map``, not ``minecraft:filled_map``?
